src/olympia/yara/models.py

```diff
@@ -0,0 +1,26 @@
+from django.db import models
+from django_extensions.db.fields.json import JSONField
+
+from olympia.amo.models import ModelBase
+from olympia.files.models import FileUpload
+
+
+class YaraResult(ModelBase):
+    upload = models.OneToOneField(FileUpload,
+                                  related_name='yara_results',
+                                  on_delete=models.CASCADE)
+    matches = JSONField(default=[])
+    version = models.OneToOneField('versions.Version',
+                                   related_name='yara_results',
+                                   on_delete=models.CASCADE,
+                                   null=True)
+
+    class Meta:
+        db_table = 'yara_results'
+
+    def add_match(self, rule, tags=[], meta={}):
```

Comment:
Don't use `{}` or `[]` (mutable objects) as defaults in Python arguments.

```python
In [2]: def append(num, numlist=[]):
   ...:     numlist.append(num)
   ...:     print(numlist)
   ...:     return numlist
   ...:

In [3]: append(5)
[5]
Out[3]: [5]

In [4]: append(7)
[5, 7]
Out[4]: [5, 7]

In [5]: append(2)
[5, 7, 2]
Out[5]: [5, 7, 2]
```

Use `None` as default values and do something like `[] if tags is not None else tags` or `tags or []` which should both work fine.

Reply:
ah, right. I keep forgetting this..